### ISSUE TITLE
NeuralNet validation not in evaluation mode.

### ISSUE THREAD

**Issue by abcsds** [skorch-dev/skorch#976]
Hi I'm doing hyper-parameter optimization on a NeuralNet, and realized that the best dropout layers are returning p=0. Checking on the evaluation step within NeuralNet I see no model.eval() call before the evaluation. This call is important for the batch normalization, and dropout layers: https://github.com/skorch-dev/skorch/blob/f69be0fa312fc78a0c386caf0422d840f68d9b1b/skorch/net.py#LL943C4-L943C4

Am I seeing this right?

**Comment by abcsds** [comment]
I saw it now: `_set_training` https://github.com/skorch-dev/skorch/blob/f69be0fa312fc78a0c386caf0422d840f68d9b1b/skorch/net.py#L909
